### DISCUSSION TITLE
plugin-cloud-storage: Update S3 URL generation

### DISCUSSION THREAD

**Discussion by yandodov** [payloadcms/payload#5591] [Feature Requests & Ideas]
Currently, the plugin generates URLs like that:

```
https://s3.us-west-1.amazonaws.com/my-bucket/my-prefix/my-image.jpg
```

Notice how the bucket is specified as part of the path. S3 also supports specifying the bucket as part of the origin, like that:

```
https://my-bucket.s3.us-west-1.amazonaws.com/my-prefix/my-image.jpg
```

I think the latter is preferred, because all URLs in the Amazon console look like that. Also, [this answer on Server Fault](https://serverfault.com/a/965940/530311) states that Amazon my sometimes redirect from the old deprecated form (that Payload uses) to the new one.

I think this should be changed:

https://github.com/payloadcms/plugin-cloud-storage/blob/e11a0fb28534b7d9fd0e91b51e7ab4df6cad664c/src/adapters/s3/generateURL.ts#L10-L14

If the new URL scheme is used, I think the `bucket` option can be dropped, because the `endpoint` will contain the bucket name already.

**Comment by dominikzogg** [comment]
@hdodov please take into account that S3 doesn't mean AWS only, so for example a self hosted minio should work as well.  So the current url generation is more generic usable.

**Comment by paulpopus** [comment]
This plugin is now being maintained in the [Packages Directory](https://github.com/payloadcms/payload/tree/main/packages/plugin-cloud-storage) of the [Payload Monorepo](https://github.com/payloadcms/payload). This repo will soon be archived and all open issues will be closed. This issue, however, will be transferred over.

**Comment by denolfe** [comment]
Moving this to feature request. We can continue the discussion there.
